openex-api/src/main/java/io/openex/service/ChallengeService.java

```diff
@@ -23,51 +25,65 @@
+    List<Inject> injects = fromIterable(this.injectRepository.findAllForChallengeId("%" + challenge.getId() + "%"));
+    List<String> exerciseIds = injects.stream().filter(i -> i.getExercise() != null).map(i -> i.getExercise().getId()).distinct().toList();
+    challenge.setExerciseIds(exerciseIds);
+    List<String> scenarioIds = injects.stream().filter(i -> i.getScenario() != null).map(i -> i.getScenario().getId()).distinct().toList();
+    challenge.setScenarioIds(scenarioIds);
+    return challenge;
+  }
 
-    @Autowired
-    public void setChallengeRepository(ChallengeRepository challengeRepository) {
-        this.challengeRepository = challengeRepository;
-    }
+  public Iterable<Challenge> getExerciseChallenges(@NotBlank final String exerciseId) {
+    Exercise exercise = exerciseRepository.findById(exerciseId).orElseThrow();
+    return resolveChallenges(exercise.getInjects())
+        .map(this::enrichChallengeWithExercisesOrScenarios)
+        .toList();
+  }
 
-    @Autowired
-    public void setExerciseRepository(ExerciseRepository exerciseRepository) {
-        this.exerciseRepository = exerciseRepository;
-    }
+  public Iterable<Challenge> getScenarioChallenges(@NotNull final Scenario scenario) {
+    return resolveChallenges(scenario.getInjects())
+        .map(this::enrichChallengeWithExercisesOrScenarios)
+        .toList();
+  }
 
-    public Challenge enrichChallengeWithExercises(Challenge challenge) {
-        List<Inject> injects = fromIterable(injectRepository
-                .findAllForChallengeId("%" + challenge.getId() + "%"));
-        List<String> exerciseIds = injects.stream().map(i -> i.getExercise().getId()).distinct().toList();
-        challenge.setExerciseIds(exerciseIds);
-        return challenge;
-    }
+  // -- PRIVATE --
 
-    public Iterable<Challenge> getExerciseChallenges(@PathVariable String exerciseId) {
-        Exercise exercise = exerciseRepository.findById(exerciseId).orElseThrow();
-        List<String> challenges = exercise.getInjects().stream()
-                .filter(inject -> inject.getContract().equals(CHALLENGE_PUBLISH))
-                .filter(inject -> inject.getContent() != null)
-                .flatMap(inject -> {
-                    try {
-                        ChallengeContent content = mapper.treeToValue(inject.getContent(), ChallengeContent.class);
-                        return content.getChallenges().stream();
-                    } catch (JsonProcessingException e) {
-                        return Stream.empty();
-                    }
-                })
-                .distinct().toList();
-        return fromIterable(challengeRepository.findAllById(challenges)).stream()
-                .map(this::enrichChallengeWithExercises).toList();
-    }
+  private Stream<Challenge> resolveChallenges(@NotNull final List<Inject> injects) {
+    List<String> challenges = injects.stream()
+        .filter(inject -> inject.getContract().equals(CHALLENGE_PUBLISH))
```

Comment:
Is better "Challenege" on the left side of equals because inject.getContract can be null? is it is null, the program should be running?